# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/modules/WebProgressChild.jsm
# Commit: 52e8c16849b3
# Full Hash: 52e8c16849b3a871a3ea6434c884d32af568337d
# Author: Barret Rennie <barret@brennie.ca>
# Date: 2019-04-04 04:33:36
# Regressor Bug: 1510569
# File Overlap Count: 1
# Description:
#   Bug 1510569 - Port onRefreshAttempted from WebProgressChild.jsm to C++ r=baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D25650
# ==============================================================================

diff -r e1fbefe38944 -r 52e8c16849b3 toolkit/modules/WebProgressChild.jsm
--- a/toolkit/modules/WebProgressChild.jsm	Wed Apr 03 17:32:41 2019 +0000
+++ b/toolkit/modules/WebProgressChild.jsm	Wed Apr 03 17:32:56 2019 +0000
@@ -27,11 +27,12 @@
 
     this.inLoadURI = false;
 
-    // NOTIFY_PROGRESS, NOTIFY_STATUS, and NOTIFY_CONTENT_BLOCKING are handled
-    // by PBrowser.
+    // NOTIFY_PROGRESS, NOTIFY_STATUS, NOTIFY_REFRESH, and
+    // NOTIFY_CONTENT_BLOCKING are handled by PBrowser.
     let notifyCode = Ci.nsIWebProgress.NOTIFY_ALL &
                         ~Ci.nsIWebProgress.NOTIFY_PROGRESS &
                         ~Ci.nsIWebProgress.NOTIFY_STATUS &
+                        ~Ci.nsIWebProgress.NOTIFY_REFRESH &
                         ~Ci.nsIWebProgress.NOTIFY_CONTENT_BLOCKING;
 
     this._filter = Cc["@mozilla.org/appshell/component/browser-status-filter;1"]
@@ -180,10 +181,6 @@
     this._send("Content:SecurityChange", json);
   }
 
-  onRefreshAttempted(aWebProgress, aURI, aDelay, aSameURI) {
-    return true;
-  }
-
   sendLoadCallResult() {
     this.mm.sendAsyncMessage("Content:LoadURIResult");
   }
@@ -191,5 +188,4 @@
 
 WebProgressChild.prototype.QueryInterface =
   ChromeUtils.generateQI(["nsIWebProgressListener",
-                          "nsIWebProgressListener2",
                           "nsISupportsWeakReference"]);
